pkg/daemon/ceph/client/filesystem.go

```diff
@@ -168,6 +169,10 @@ func AddDataPoolToFilesystem(context *clusterd.Context, clusterInfo *ClusterInfo
 	args := []string{"fs", "add_data_pool", name, poolName}
 	_, err := NewCephCommand(context, clusterInfo, args).Run()
 	if err != nil {
+		// TODO: remove this once https://github.com/rook/rook/issues/12673 is fixed.
```

Comment:
This is the expected error from Reef, not sure we'll have any other fix than this.
```suggestion
		// Reef disallows calling add_data_pool for a pool that has already been added, so ignore the error code.
		// Previous releases do not return an error when an existing data pool is added.
```